File: AdityaKumar1010707/BinarySearch/hIndexII.cpp
class Solution {
    public:
        int hIndex(vector<int>& arr) {
    
            sort(arr.begin(),arr.end());
    
            int count  = 0;
    
            int n = arr.size();
    
            int ans = 0;
    
            for(int i=n-1;i>=0;i--){
                if(arr[i]>0) count++;
                
                int elements = arr.size()-i;
                
                if(count == elements && arr[i] >=count) ans = count;
            }
    
            return ans;
            
        }
    };

File: Amit-Kumar-Singh/Binary_Search/H_Index_II.java
class Solution {
    public int hIndex(int[] citations) {
        int n =citations.length;
        int start = 0;
        int end = n-1;
        while(start<=end){
            int mid = (start+end)/2;
            if(citations[mid]==n-mid){
                return citations[mid];
            }
            else if(citations[mid]<n-mid){
                start = mid+1;
            }
            else{
                end = mid-1;
            }
        }
        return n-start;
    }
}